src/main/java/catbot/io/CatbotConsoleIO.java

```diff
@@ -0,0 +1,229 @@
+        while (true) {
+            nextLine = nextLine();
+            namedParameterMap = parser.parse(nextLine);
+            if (namedParameterMap.keySet().size() == 1) {
+                for (String command : namedParameterMap.keySet()) {
+                    return new CommandArgument(command, namedParameterMap.get(command));
+                }
+            }
+        }
+    }
+
+    //endregion
+
+    //region ErrorIndicatorIo
+
+    @Override
+    public void indicateInvalidCommand(String attemptedCommand) {
+        warn("idgi ;-;");
+    }
+
+    @Override
+    public void indicateInvalidInteger(String attemptedInteger) {
+        warn("that doesn't look like a number... number pls");
+    }
+
+    @Override
+    public void indicateInvalidIndex(int attemptedIndex, TaskList.Bounds bounds) {
+        warn("i expected a number from " + bounds.lowerBound + " to " + bounds.upperBound + "...");
+    }
+
+    @Override
+    public void indicateArgumentInvalid(InvalidState invalidState, NamedParameterMap namedParameterMap) {
+        switch (invalidState) {
+            case PARAMETER_EMPTY:
+                for (String arg : namedParameterMap.keySet()) {
+                    warn(arg + " is empty");
+                }
+                send("please make sure these arguments are filled!");
+                break;
+            case PARAMETER_MISSING:
+                for (String arg : namedParameterMap.keySet()) {
+                    warn(arg + " is missing");
+                }
+                send("please make sure to include them next time!");
+                break;
+            case NOT_A_DATE:
+                for (String arg : namedParameterMap.keySet()) {
+                    warn(arg + " is set to \"" + namedParameterMap.get(arg) + "\", which is not a date!");
+                }
+                break;
+            default:
+                throw new RuntimeException();
+        }
+    }
+
+    //endregion
+
+    //region TaskAssistantIo
+
+    @Override
+    public void printTaskList(TaskList taskList) {
+        int i = 1;
+        int intlen = 0;
+        for (int len = taskList.size(); len>0; intlen++) len/=10;
+        for (Task t : taskList.getTasks()) {
+            send(String.format("%" + intlen + "d", i++) + ". " + t);
+        }
+    }
+
+    @Override
+    public void printTaskAdded(TaskList taskList) {
+        int index = taskList.size()-1;
+        send("Added: " + (index+1) + ". " + taskList.getTask(index));
+    }
+
+    @Override
+    public void printTaskDeleted(Task deleted) {
+        send("Deleted: " + deleted);
+    }
+
+    @Override
+    public void printTaskModified(TaskList taskList, int index) {
+        send((index+1) + ". " + taskList.getTask(index));
+    }
+    //endregion
+
+    //region Internal Helper
+
+    private String format(String s, String prefix) {
+        if (s == null) return "";
+        return s.replaceAll("(^|\n)", "$1"+prefix);
+    }
+
+    private void line() {
```

Comment:
Perhaps adding a verb in the function name might be more intuitive?

Reply:
Good catch; I'll do that!